### ISSUE TITLE
Borders overlap on tiled layouts when titlebar is hidden

### ISSUE THREAD

**Issue by b01t** [awesomeWM/awesome#2131]
Output of `awesome --version`:

awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✔
 • LGI version: 0.9.1

awesome v4.2 (Human after all)
 • Compiled against Lua 5.3.3 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.5
 • LGI version: 0.9.2

**How to reproduce the issue:**

* Use the default theme
* Set any tiled layout
* Notice borders don't overlap
* Disable titlebars in the rc.lua rules section
* Notice titlebars now overlap in several places

**Actual result:**

Overlapping borders

**Expected result:**

No overlapping borders

![awesome_vm-titlebar_off-1](https://user-images.githubusercontent.com/4450379/33530010-56a00212-d879-11e7-92ea-51e76b7bb7aa.png)
![awesome_vm-titlebar_off-2](https://user-images.githubusercontent.com/4450379/33530011-56c892f4-d879-11e7-9e40-58d471f933a7.png)
![awesome_vm-titlebar_on-1](https://user-images.githubusercontent.com/4450379/33530012-56e7b4cc-d879-11e7-9465-d8addd743b6f.png)
![awesome_vm-titlebar_on-2](https://user-images.githubusercontent.com/4450379/33530013-57074508-d879-11e7-8ae9-20ca6bce181f.png)

**Comment by Elv13** [comment]
Can you try with size honoring disabled (see the FAQ)? It's still a bug, but it will probably mitigate it.

**Comment by b01t** [comment]
Size hint honoring disabled (made a rule for instance:x-terminal-emulator) does indeed mitigate the issue. Kind of looks cool too, at least for terminals.

**Comment by Elv13** [comment]
Yeah, I was kind of aware of the problem, but nobody complained before literally 2 user filled a bug on the same day (#2131 #2132). Size hints are really the last big area where we still lack full unit test coverage.
